### PR TITLE
Improving code quality

### LINKED ISSUE #673: Improving the quality of the code
At the time of creating this issue, the score of pylint on the entire Python code for only errors is 9.97/10 (--disable=typecheck,W,C,R $(git ls-files '*.py')).
The score for only warnings is 9.19/10 (--disable=typecheck,E,C,R $(git ls-files '*.py')).

We currently only check the score of error in our CI/CD. I think we should improve the score for error to 10, and check the warning score to stay above 9. After this improvement, the check in pylint.yml should change to:

pylint --fail-under=10 --disable=typecheck,W,C,R $(git ls-files '*.py')
pylint --fail-under=9 --disable=typecheck,E,C,R $(git ls-files '*.py')

### PR DESCRIPTION
**Pull request by afard**
All errors reported by pylint are resolved. Let's keep it that way. Therefore, I change the passing score for error to 10, and passing score of warning above 9.

This PR resolves #673.

### CODE REVIEW

File: verticapy/core/vdataframe/_math.py [hunk 1]
```diff
@@ -43,20 +43,20 @@ class vDFMath(vDFFilter):
...
+    def __ceil__(self) -> "vDataFrame":
         vdf = self.copy()
         columns = vdf.numcol()
         for col in columns:
             if vdf[col].category() == "float":
-                vdf[col].apply_fun(func="ceil", x=n)
+                vdf[col].apply_fun(func="ceil")
         return vdf
 
-    def __floor__(self, n: int) -> "vDataFrame":
```
Comment (oualib) [inline verticapy/core/vdataframe/_math.py:54]: why those changes?

**Review by oualib — COMMENTED** [review]
can you please explain your changes. I do not understand some of them.

File: .github/workflows/pylint.yml [hunk 2]
```diff
@@ -21,6 +21,10 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install pylint
+        pip install .
+        pip install -r requirements-dev.txt
+        pip install -r requirements-testing.txt
```
Comment (afard) [inline .github/workflows/pylint.yml:26]: The pylint test checks all imports in the entire Python code. Therefore, we need to install all of them before running pylint.

File: .github/workflows/pylint.yml [hunk 3]
```diff
@@ -21,6 +21,10 @@ jobs:
...
     - name: Analysing the code with pylint
       run: |
-        pylint --fail-under=9 --disable=typecheck,W,C,R $(git ls-files '*.py')
+        pylint --fail-under=10 --disable=typecheck,W,C,R --extension-pkg-allow-list=scipy.special $(git ls-files '*.py')
```
Comment (afard) [inline .github/workflows/pylint.yml:29]: We increase the passing rate for errors to 10 out of 10.

File: .github/workflows/pylint.yml [hunk 4]
```diff
@@ -21,6 +21,10 @@ jobs:
...
+        pylint --fail-under=9 --disable=typecheck,E,C,R $(git ls-files '*.py')
```
Comment (afard) [inline .github/workflows/pylint.yml:30]: We add a limit on warnings to be above 9 (out of 10)

File: requirements-testing.txt
Comment (afard) [inline requirements-testing.txt:1]: The dependency list in tox.ini are moved here so that they can be called from .github/workflows/pylint.yml.

File: verticapy/core/vdataframe/_corr.py [hunk 5]
```diff
@@ -1028,6 +1028,9 @@ def corr_pvalue(
             )
             x = val * val * n * min(k, r)
             pvalue = scipy_st.chi2.sf(x, (k - 1) * (r - 1))
+        else:
+            raise ValueError(f"{method} is not a valid method.")
```
Comment (afard) [inline verticapy/core/vdataframe/_corr.py:1032]: There was a pylint error complaining that  pvalue might not have been initialized.

File: verticapy/core/vdataframe/_math.py (hunk 1 above)
Comment (afard) [reply verticapy/core/vdataframe/_math.py:54]: These are called dunder or magic methods (https://www.geeksforgeeks.org/dunder-magic-methods-python/).
It is convention that when an object implements __ceil__() function it actually implements a behavior for math.ceil(obj). That function doesn't accept any extra argument. Besides, what is the use of a precision argument for ceil and floor functions?
https://blog.finxter.com/python-__ceil__-magic-method
https://mathspp.com/blog/pydonts/dunder-methods

File: verticapy/core/vdataframe/base.py [hunk 6]
```diff
@@ -195,10 +195,12 @@ def __init__(
                 )
 
         if isinstance(input_relation, (TableSample, list, np.ndarray, dict)):
-            return self._from_object(input_relation, usecols)
```
Comment (afard) [inline verticapy/core/vdataframe/base.py:198]: The __init__ function is not supposed to return anything although the self._from_object function None.

File: verticapy/core/vdataframe/base.py [hunk 7]
```diff
@@ -324,7 +326,7 @@ def _from_object(
                 tb_final[col] = tb[col]
             tb = TableSample(tb_final)
 
-        return self.__init__(input_relation=tb.to_sql())
```
Comment (afard) [inline verticapy/core/vdataframe/base.py:327]: `return` is really redundant when `__init__` is not supposed to return anything.

File: verticapy/datasets/loaders.py [hunk 8]
```diff
@@ -85,7 +85,6 @@ def load_dataset(
                 query = query.format(f"LOCAL '{path}'")
                 _executeSQL(query, title="Ingesting the data.")
 
-            _executeSQL("COMMIT;", title="Commit.")
```
Comment (afard) [inline verticapy/datasets/loaders.py:88]: This is not related to the pylint test. I just found the warning message after loading datasets annoying.
"`COPY` automatically commits itself and any current transaction except when loading temporary tables." It doesn't seem that we load any temporary tables here, do we?

File: verticapy/machine_learning/vertica/automl/supervised.py [hunk 9]
```diff
@@ -447,7 +447,7 @@ def fit(
                     ]
             else:
                 raise ValueError(
-                    f"Parameter 'estimator_type' must be in auto|binary|multi|regressor. Found {estimator_type}."
+                    f"Parameter 'estimator_type' must be in auto|binary|multi|regressor. Found {self.parameters['estimator_type']}."
```
Comment (afard) [inline verticapy/machine_learning/vertica/automl/supervised.py:450]: There was no `estimator_type` variable defined.

File: verticapy/machine_learning/vertica/base.py [hunk 10]
```diff
@@ -1001,10 +1002,7 @@ def _compute_features_importance(self, tree_id: Optional[int] = None) -> None:
         )
         importance = self._format_vector(self.X, importance)
         if isinstance(tree_id, int) and (0 <= tree_id < self.n_estimators_):
-            if hasattr(self, "features_importance_trees_"):
-                self.features_importance_trees_[tree_id] = importance
```
Comment (afard) [inline verticapy/machine_learning/vertica/base.py:1005]: The pylint test was complaining that self.features_importance_trees_ is used before it is defined.

File: verticapy/plotting/_matplotlib/animated/bar.py [hunk 11]
```diff
@@ -159,14 +159,14 @@ def animate(i: int) -> tuple[Axes]:
 
         return animate
 
-    def _compute_anim_params(
-        self, date_f: Optional[Callable] = None, **style_kwargs
-    ) -> tuple:
+    def _compute_anim_params(self, date_f: Optional[Callable], **style_kwargs) -> tuple:
         if isinstance(date_f, NoneType):
 
-            def date_f(x: Any) -> str:
```
Comment (afard) [inline verticapy/plotting/_matplotlib/animated/bar.py:167]: The pylint test was complaining that the `date_f` was already defined.

File: verticapy/sdk/vertica/udf/utils.py [hunk 12]
```diff
@@ -92,7 +92,7 @@ def get_set_add_function(
     elif ftype == (datetime.datetime, datetime.tzinfo):
         return f"{func}TimestampTz"
     else:
-        raise (
+        raise ValueError(
```
Comment (afard) [inline verticapy/sdk/vertica/udf/utils.py:95]: The class of a raised exception should be explicitly defined.

File: verticapy/tests/stats/test_stats.py [hunk 13]
```diff
@@ -336,12 +330,6 @@ def test_regexp_substr(self, titanic_vd):
             == "REGEXP_SUBSTR(\"name\", '([A-Za-z])+\\. ', 1, 1)"
         )
 
-    def test_regexp_count(self, titanic_vd):
```
Comment (afard) [inline verticapy/tests/stats/test_stats.py:339]: The exact test was repeated!

File: verticapy/tests/vDataFrame/test_vDF_plot_plotly.py [hunk 14]
```diff
@@ -777,14 +777,14 @@ def test_properties_yaxis_title_for_partitioned_data(
...
 
-    def test_properties_orientation(self, load_plotly, dummy_dist_vd):
+    def test_properties_orientation_default(self, load_plotly, dummy_dist_vd):
         # Arrange
         # Act
         result = dummy_dist_vd["0"].boxplot()
         # Assert
         assert result.data[0]["orientation"] == "h", "Orientation is not correct"
 
-    def test_properties_orientation(self, load_plotly, dummy_dist_vd):
```
Comment (afard) [inline verticapy/tests/vDataFrame/test_vDF_plot_plotly.py:787]: Two tests had the same names.

File: verticapy/tests/vDataFrame/test_vDF_plot_plotly.py [hunk 15]
```diff
@@ -912,7 +912,7 @@ def test_data_median_value_for_partitioned_data_for_x_is_0(
             50, 1
         ), "median not computed correctly for binary=0"
 
-    def test_data_median_value_for_partitioned_data_for_x_is_0(
+    def test_data_median_value_for_partitioned_data_for_x_is_1(
```
Comment (afard) [inline verticapy/tests/vDataFrame/test_vDF_plot_plotly.py:915]: Two tests had the same names.

File: verticapy/tests/vDataFrame/test_vDF_plot_plotly.py [hunk 16]
```diff
@@ -1415,7 +1415,7 @@ def test_properties_xaxis(self, load_plotly, dummy_date_vd):
             result.layout["xaxis"]["title"]["text"] == column_name
         ), "X axis label incorrect"
 
-    def test_properties_xaxis(self, load_plotly, dummy_date_vd):
+    def test_properties_yaxis(self, load_plotly, dummy_date_vd):
```
Comment (afard) [inline verticapy/tests/vDataFrame/test_vDF_plot_plotly.py:1418]: Two tests had the same names.

File: verticapy/tests/vModel/test_mca.py [hunk 17]
```diff
@@ -121,7 +121,7 @@ def test_plot(self, model):
         result = model.plot(dimensions=(2, 3))
         assert len(result.get_default_bbox_extra_artists()) == 8
 
-    def test_plot_contrib(self, model):
+    def test_plot_var(self, model):
```
Comment (afard) [inline verticapy/tests/vModel/test_mca.py:124]: Two tests had the same names.

File: verticapy/tests_new/plotting/base_test_files.py [hunk 18]
```diff
@@ -300,12 +307,6 @@ def create_plot(self):
             {"columns": self.COL_NAME_VDF_1},
         )
 
-    def col_name_param():
```
Comment (afard) [inline verticapy/tests_new/plotting/base_test_files.py:303]: There was a pylint error as follows:
`verticapy/tests_new/plotting/base_test_files.py:303:4: E0211: Method 'col_name_param' has no argument (no-method-argument)`
Adding `@staticmethod` decorator helped with pylint, but caused failure at the time of running the unit tests. The easiest fix was moving the function out of the class scope.

**Review by oualib — APPROVED** [review]
